### PR TITLE
add minimum issue date to the metadata with some tests(merge to dev)

### PR DESCRIPTION
**Pull request by zhuoran-Cheng16**
merge main to dev and delete the covid-hosp part

### CODE REVIEW

File: src/acquisition/covidcast/database.py
Comment (melange396) [inline src/acquisition/covidcast/database.py:303]:
```suggestion
      UNION

      SELECT
```

File: src/acquisition/covidcast/database.py
Comment (melange396) [inline src/acquisition/covidcast/database.py:317]: why did you do these in descending order?

**Review by melange396 — CHANGES_REQUESTED** [review]
i think you forgot some changes to https://github.com/cmu-delphi/delphi-epidata/blob/main/integrations/acquisition/covidcast/test_covidcast_meta_caching.py ?

File: src/acquisition/covidcast/database.py
Comment (zhuoran-Cheng16) [reply src/acquisition/covidcast/database.py:317]: Oh I think because with the DESC order the first date will be the min-issue date, and with ASC the first date will be the max-issue date. But they are same because they use min() and max() to find the date, I will change it back~

**Review by melange396 — APPROVED** [review]
looks pretty good to me!  ...if you wanted to add some more test cases, that might be nice

File: src/acquisition/covidcast/signal_dash_data_generator.py
Comment (zhuoran-Cheng16) [reply src/acquisition/covidcast/signal_dash_data_generator.py:187]: Yes it been used in delphi-epidata/tests/acquisition/covidcast/test_signal_dash_data_generator.py to test if it can get the earliest issue from metadata.

Comment (krivard) [reply src/acquisition/covidcast/signal_dash_data_generator.py:187]: hmmm if the only use of get_earliest_issue_from_metadata is the test of get_earliest_issue_from_metadata, then we should probably drop it. unless you are planning to use get_earliest_issue_from_metadata in a future PR?

Comment (zhuoran-Cheng16) [reply src/acquisition/covidcast/signal_dash_data_generator.py:187]: Gotcha! I don't think it will be use in future...I will drop it now.

**Comment by krivard** [comment]
Would you update the PR description summarizing what this PR changes? That way if we ever need to go back and find this PR again we'll know what we're looking at

**Comment by melange396** [comment]
this commit ( https://github.com/cmu-delphi/delphi-epidata/commit/14999a3a7b94c6ae7a52ac9dc1cdea1904d2aa73 ) looks great, but it should probably be part of its own PR

**Comment by krivard** [comment]
Whoops, that's my mistake, that should not have been possible. Fixed now for future PRs.

**Comment by zhuoran-Cheng16** [comment]
> Whoops, that's my mistake, that should not have been possible. Fixed now for future PRs.

Sorry Katie, it's my mistake, I just push the SQL instance part to my fork respo in order to chat with Tara and made some modified with her. After we changes together, now we have a new version of the SQL instance, can I reopen this pr and update the part to new version?

**Comment by zhuoran-Cheng16** [comment]
> Would you update the PR description summarizing what this PR changes? That way if we ever need to go back and find this PR again we'll know what we're looking at

Ok I already update the top comment to summarize the changes of this PR.

**Comment by krivard** [comment]
Summary is good! thanks.

> can I reopen this pr

Alas, no -- GitHub doesn't support amending PRs which have already been merged. To modify the SQL queries further, open a new PR.

**Comment by krivard** [comment]
This PR failed to build once merged with [a pile of test failures](https://gist.github.com/krivard/f304d414fa62acfe5a676fb837662421); probably due to untested SQL query modifications. I'm backing it out so we can release epidata this morning.

Next steps for @zhuoran-Cheng16, once the release is complete:
* make a new branch from cmu-delphi:dev
* cherry-pick everything except 14999a3
* submit a new PR (& use a branch, not a fork, so that any failing CI tests block merge)

**Pull request merged**
